src/core/types/secret_bytes.rs

```diff
@@ -0,0 +1,136 @@
+use crate::prelude::*;
+
+#[macro_export]
+macro_rules! decl_secret_bytes {
+    (
+        $(
+            #[doc = $expr: expr]
+        )*
+        $struct_name: ident,
+        $byte_count: literal
+    ) => {
+        paste! {
+
+            $(
+                #[doc = $expr]
+            )*
+            #[derive(Zeroize, ZeroizeOnDrop, derive_more::Debug, derive_more::Display)]
+            #[debug("OBFUSCATED")]
+            #[display("OBFUSCATED")]
+            pub struct $struct_name(Box<[u8; Self::LENGTH]>);
```

Comment:
A newtype around bytes of length `$byte_count` (input to macro) - which is `Zeroize` and `ZeroizeOnDrop`.